pdl-compiler/src/backends/rust/parser.rs

```diff
@@ -354,17 +367,20 @@ impl<'a> FieldParser<'a> {
         // given via a static count, a count field, a size field, or
         // unknown.
         enum ArrayShape {
-            Static(usize),                  // Static count
-            CountField(proc_macro2::Ident), // Count based on count field
-            SizeField(proc_macro2::Ident),  // Count based on size and field
-            Unknown,                        // Variable count based on remaining bytes
+            Static(usize),                        // Static count
+            CountField(proc_macro2::Ident),       // Count based on count field
+            SizeField(proc_macro2::Ident),        // Count based on size and field
+            ElementSizeField(proc_macro2::Ident), // Count based on element size and field
```

Comment:
Element size does not represent the size of the array.
You could technically have both:

```
struct Test {
    _element_size_(foo) : 8,
    _size_(foo): 8,
    foo: Foo[],
}
```
Element size should be added to `ElementWidth` instead.